libcodechecker/libclient/authentication_helper.py

```diff
@@ -66,6 +69,8 @@ def wrapper(self, *args, **kwargs):
             except TProtocolException as ex:
                 print("Connection failed to {0}:{1}"
                       .format(self.__host, self.__port))
+                import traceback
```

Comment:
This trace goes out to the user in the command line?

Reply:
It does, but this is most needed to help debugging later on, from the users' perspective. We cannot always recreate every scenario. A `Connection failed` is a useless error message, because a `TProtocolException` can indicate stuff like encoding issues, where **a connection is properly made**, but the communication flipped itself.

This traceback *SHOULD NOT* go to the user. It only happens if a serious Thrift error is thrown, such as forcing the authentication client onto a bad API version, or a bad endpoint. If the API clients are used properly in the code, the API versioning changes are handled by other exception handlers a lot more gracefully.